### DISCUSSION TITLE
Versel Serverless Function 4.5MB body size limit

### DISCUSSION THREAD

**Discussion by amroczeK** [vercel/community#2454] [Help] [answered]
### Summary

I have developed a complex form that allows users to attach photos to the form which gets encoded to base64. The form is to make a service request for their wheelchairs/mobility equipment, which requires photo attachment of the damage.

On form submission, the form data gets converted to a HTML template and I use the Vercel Serverless functions to make an API request to **SendGrid** with the attachments. Sendgrid will then send the email to the specified user with the template and attached images.

The problem I am facing is that photos users take are normally from their phones and are generally >1mb in size. So we will quickly reach the 4.5mb limit if a user attaches more than 1 large image. Returning `413: FUNCTION_PAYLOAD_TOO_LARGE`

https://vercel.com/guides/how-to-bypass-vercel-body-size-limit-serverless-functions

**Question:** Do the other available plans increase this 4.5 mb limit or is this the same across all plans? I am currently on the Hobby plan.

Alternatively I might have to compress the images on the client side before sending the form.

**Comment by amyegan** [comment, marked as answer]
The 4.5MB limit is the same for all plans. Compressing the images would be good option allowing you to keep using your existing function. I'm not sure if Edge Functions would work in this case, since they have their own [limitations](https://vercel.com/docs/concepts/functions/edge-functions/limitations#) to consider, but [streaming data in Edge Functions](https://vercel.com/docs/concepts/functions/edge-functions/streaming#) could be another solution.

You can find more info on account limits, including a [functions comparison](https://vercel.com/docs/concepts/limits/overview#functions-comparison) table, at https://vercel.com/docs/concepts/limits/overview
